### PR TITLE
[vividus-plugin-lighthouse] Add ability to compare lighthouse reports

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: docs/modules/plugins/pages/plugin-lighthouse.adoc [hunk 1]
```diff
@@ -79,3 +79,40 @@ When I perform Lighthouse full scan of `https://dequeuniversity.com/demo/mars` p
...
+----
+
+* `$scanType` - The scan type to use:
+** `full` - The scan is performed on both `desktop` and `mobile` devices.
+** `desktop` - The scan is performed on `desktop` device.
+** `mobile` - The scan is performed on `mobile` device.
+* `$checkpointPage` - The checkpoint page.
+* `$baselinePage` - The baseline page.
+
+.Compare Lighthouse results of two sites
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-lighthouse.adoc:114]:
```suggestion
.Compare Lighthouse results of two pages
```

File: vividus-plugin-lighthouse/src/main/java/org/vividus/lighthouse/LighthouseSteps.java [hunk 2]
```diff
@@ -130,6 +135,71 @@ public void performLighthouseScan(ScanType scanType, String webPageUrl,
...
+            Map<String, Integer> checkpointScores = getCategoryScores(checkpoint);
+
+            softAssert.runIgnoringTestFailFast(() -> getCategoryScores(baseline).forEach((categoryKey, baselineScore) ->
+            {
+                Integer checkpointScore = checkpointScores.get(categoryKey);
+
+                if (checkpointScore >= baselineScore)
+                {
+                    softAssert
+                            .recordPassedAssertion(String.format("[%s] The %s audit is passed", strategy, categoryKey));
```
Comment (valfirst) [inline vividus-plugin-lighthouse/src/main/java/org/vividus/lighthouse/LighthouseSteps.java:166]: put some score values to make assertion more informative

**Review by valfirst — CHANGES_REQUESTED** [review]

File: docs/modules/plugins/pages/plugin-lighthouse.adoc [hunk 3]
```diff
@@ -79,3 +79,40 @@ When I perform Lighthouse full scan of `https://dequeuniversity.com/demo/mars` p
...
+
+[source,gherkin]
+----
+Then Lighthouse $scanType audit scores for `$checkpointPage` page are not less than for `$baselinePage` page
+----
+
+* `$scanType` - The scan type to use:
+** `full` - The scan is performed on both `desktop` and `mobile` devices.
+** `desktop` - The scan is performed on `desktop` device.
+** `mobile` - The scan is performed on `mobile` device.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-lighthouse.adoc:110]: I guess this part is duplicated

File: vividus-plugin-lighthouse/src/main/java/org/vividus/lighthouse/LighthouseSteps.java [hunk 4]
```diff
@@ -130,6 +135,74 @@ public void performLighthouseScan(ScanType scanType, String webPageUrl,
...
+                            strategy, categoryKey, checkpointScore, acceptableScorePercentageDelta, baselineScore));
+                    return;
+                }
+
+                softAssert.recordFailedAssertion(String.format("[%s] The %s audit score is decreased on %d percents",
+                        strategy, categoryKey, scoreDecrease));
+            }));
+
+            attachmentPublisher.publishAttachment(
+                    "/allure-customization/webjars/vividus-lighthouse-viewer-adaptation/index.html",
```
Comment (valfirst) [inline vividus-plugin-lighthouse/src/main/java/org/vividus/lighthouse/LighthouseSteps.java:188]: shouldn't it be FTL?

Comment (uarlouski) [reply vividus-plugin-lighthouse/src/main/java/org/vividus/lighthouse/LighthouseSteps.java:188]: not required

Comment (uarlouski) [reply vividus-plugin-lighthouse/src/main/java/org/vividus/lighthouse/LighthouseSteps.java:188]: just didn't want to add more actions to the build guide, only essential ones

**Review by draker94 — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**